### DISCUSSION TITLE
Add ability to expose authelia as an LDAP server

### DISCUSSION THREAD

**Discussion by prabirshrestha** [authelia/authelia#4620] [Ideas]
### Description

I would like to be able to expose authelia as an LDAP server so that I don't need to maintain yet another tool. [gldap](https://github.com/jimlambrt/gldap) library could be used to provide ldap server within authelia.

**Comment by james-d-elliott** [comment]
Hi @prabirshrestha we're looking into a few things at the moment. One thing that we've been thinking about for some time is [SCIM 2.0](http://www.simplecloud.info/).  It's conceivable if we were to implement [SCIM 2.0](http://www.simplecloud.info/) that we'd make this service a separate service, and that this Authelia would need *some way* to communicate with this service.

I believe [github.com/jimlambrt/gldap](https://github.com/jimlambrt/gldap) or [github.com/nmcclain/ldap](https://github.com/nmcclain/ldap) both offer the ability to achieve this. I'm curious if you had any particular knowledge around either of these libraries and which one is better? I lean towards the one you suggested since it uses common libraries we already use (and I'm a contributor of).

**Comment by prabirshrestha** [comment]
I'm not familiar with SCIM but briefly looking at it, it does seem to satisfy the needs in case it needs to extends to LDAP. The reason I prefer LDAP is that most apps support it out of the box without asking them to implement SCIM.

I don't have experience with both the ldap libraries.

My use case is to be able to use sqlite storage backend of authelia so it stores the user and password information. I can then move this sqlite file from home to cloud to sync user information. I like how it is a single binary with one docker image to configure authelia.

**Comment by james-d-elliott** [comment]
Thanks for submitting this feature request, we greatly appreciate ideas like this from the community. We've decided to for the time being convert this issue to an idea discussion for the following reasons:

- This idea has __*NOT*__ been rejected/declined/etc.
- This prevents our issue list being obscure due to excessive items that we have no direct plans to implement at this stage.
- This prevents this reasonable idea from being lost if it were closed.
- This allows people to vote on the idea.
